src/tf_keyboard_cal_gui.h

```diff
@@ -0,0 +1,236 @@
+/****************************************************************************************************
+ *  Software License Agreement (BSD License)
+ *  
+ *  Copyright 2017, Andy McEvoy
+ *  
+ *  Redistribution and use in source and binary forms, with or without modification, are permitted
+ *  provided that the following conditions are met:
+ *  
+ *  1. Redistributions of source code must retain the above copyright notice, this list of conditions
+ *  and the following disclaimer.
+ *  
+ *  2. Redistributions in binary form must reproduce the above copyright notice, this list of
+ *  conditions and the following disclaimer in the documentation and/or other materials provided
+ *  with the distribution.
+ *  
+ *  3. Neither the name of the copyright holder nor the names of its contributors may be used to
+ *  endorse or promote products derived from this software without specific prior written permission.
+ *  
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR
+ *  IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND
+ *  FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR
+ *  CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ *  DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ *  DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER
+ *  IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT
+ *  OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ ****************************************************************************************************/
+
+/*
+ * Author(s) : Andy McEvoy ( mcevoy.andy@gmail.com )
+ * Desc      : Rviz display panel for dynamically manipulating TFs
+ * Created   : 25 - April - 2017
+ */
+
+#ifndef TF_KEYBOARD_CAL_GUI_H
+#define TF_KEYBOARD_CAL_GUI_H
+
+#ifndef Q_MOC_RUN
+#include <ros/ros.h>
+#include <ros/package.h>
+#include <rviz/panel.h>
+#endif
+
+#include <tf_keyboard_cal/tf_remote_receiver.h>
+
+#include <QTabWidget>
+#include <QtGui>
+#include <QPushButton>
+#include <QLabel>
+#include <QCheckBox>
+#include <QLineEdit>
+#include <QComboBox>
+#include <QFrame>
+#include <QDoubleValidator>
+#include <QDir>
+#include <QHBoxLayout>
+#include <QVBoxLayout>
+#include <QGroupBox>
+#include <QFileDialog>
+#include <QGridLayout>
+
+#include <tf/transform_datatypes.h>
+
+#include <interactive_markers/interactive_marker_server.h>
+#include <interactive_markers/menu_handler.h>
+
+#include <Eigen/Geometry>
+#include <eigen_conversions/eigen_msg.h>
+
+#include <iostream>
+#include <sstream>
+#include <fstream>
+#include <stdio.h>
+
+#include <boost/algorithm/string.hpp>
+
+namespace tf_keyboard_cal
+{
+struct tf_data{
+  std::size_t id_;
```

Comment:
it's used. I keep it there to generate a unique tf even if the user supplies the same names for "to" and "from"